src/engine/strat_engine/thinpooldev.rs

```diff
@@ -0,0 +1,88 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+use devicemapper::{DM, DevId, DeviceInfo, DmFlags};
+use engine::EngineResult;
+
+use std::path::PathBuf;
+
+use super::blockdev::BlockDev;
+
+use types::DataBlocks;
+use types::Sectors;
+
+pub struct ThinPoolDev {
```

Comment:
Whatever you know that I don't. For example, what does low-water-mark really mean? I know what it means when I'm at the shore, but that's not really enough for me to figure out what it means in this context.